PowerValidatedSolutions.psm1

```diff
@@ -20714,6 +20721,384 @@ Function Get-PasswordPolicyConfig {
+    $defaultSection = "passwordExpiration", "passwordComplexity", "accountLockout"
+	
+    foreach ($product in $customProductList) {
+        if (-Not $defaultProductList.Name.Contains($product.Name)) {
+            Write-LogMessage -Type ERROR -Message "Unknow product type $($product.Name), please check the custom configuration file and run the cmdlet again" -Colour Red
+            $encounterError = "True"
+            break
+        }
+        # Validating product sections in customConfig file
+        if ($encounterError -ne "True") {
+            $customSectionList = $customConfig.($product.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+            foreach ($section in $customSectionList) {
+                 if (-Not $defaultSection.Contains($section.Name)) {
+                    Write-LogMessage -Type ERROR -Message "Unknow section $($section.Name) under product $($product.Name), please check the custom configuration file and run the cmdlet again" -Colour Red
+                    $encounterError = "True"
+                    break
+                }
+                # Validate parameters in customConfig file
+			    if ($encounterError -ne "True") {
+				    $defaultParameterList = $defaultConfig.($product.Name).($section.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+			    	$customParameterList = $customConfig.($product.Name).($section.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+			    	foreach ( $parameterName in $customParameterList) {
+			    		if ( -Not $defaultParameterList.Name.Contains($parameterName.Name)) {
+			    			Write-LogMessage -Type ERROR -Message "Unknow parameter $($parameterName.Name) under $($section.Name) under product $($product.Name), please check the custom configuration file and run the cmdlet again" -Colour Red
+			    			$encounterError = "True"
+			    			break
+			    		} elseif ($customConfig.($product.Name).($section.Name).($parameterName.Name) -eq "" -and $parameterName.Name -ine "email") {
+			    			Write-LogMessage -Type ERROR -Message "$($product.Name):$($section.Name):$($parameterName.Name) parameter is not configured, please check the custom password policy json file" -Colour Red
+			    			$encounterError = "True"
+			    			break
+			    		}
+			    	}
+			    }
+            }
+        }
+    }
+
+    # Validating parameter values
+    if ($encounterError -ne "True") {
+        foreach ($product in $customProductList) {
+            $customSectionList = $customConfig.($product.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+            foreach ($section in $customSectionList) {
+                $defaultParameterList = $defaultConfig.($product.Name).($section.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
```

Comment:
Change: where to Where-Object

Reply:
resolved in 2948bc273bae34312514db8aba5d7cb584efd8d4